grub-core/loader/efi/chainloader.c

```diff
@@ -250,6 +251,57 @@ make_file_path (grub_efi_device_path_t *dp, const char *filename)
   return file_path;
 }
 
+/* Clear unused memory.  This is meant to make measurements of DLME more
+ * deterministic in the face of firmware not bothering to zero area containing
+ * padding. */
```

Comment:
This means one of two things:
- DRTM measures padding that isn't used by Xen
- Xen uses memory that it doesn't initialize

We should at least mention which one it is as a known limitation.

Reply:
This is for padding.  Xen can use uninitialized memory, but I don't know that it does, so not mentioning it.

https://github.com/TrenchBoot/grub/compare/f9b8f78b2fd7d2e945b53ddb3998c66a90cbbaed..a8d47f8f8e0f43fb8ca6276489ec74e83b4a78b0#diff-a532ce1960f728d4d196008e9beb0a0610b9aa7088dadfb233dc6c9b1ff1d782R404-R411

Also cleaned up code a bit.